OpenSim/Common/Component.cpp

```diff
@@ -1526,11 +1529,39 @@ void Component::dumpConnections() const {
 
 
 void Component::initComponentTreeTraversal(const Component &root) const {
-    // Going down the tree, node is followed by all its
-    // children in order, last child's successor is the parent's successor.
+    // Going down the tree, this node is followed by all its children.
+    // The last child's successor (next) is the parent's successor.
+
+    const size_t nmsc = _memberSubcomponents.size();
+    const size_t npsc = _propertySubcomponents.size();
+    const size_t nasc = _adoptedSubcomponents.size();
+
+    // If this isn't the root component and it has no parent, then
+    // this is an orphan component and we likely failed to call 
+    // finalizeFromProperties on the root OR this is a clone that
+    // has not been added to the root (in which case would have a parent).
+    if ((this != &root) && !this->hasParent() ) {
+        OPENSIM_THROW(ComponentIsAnOrphan, getName(), getConcreteClassName());
```

Comment:
> My understanding is that we generally want to flag orphaned components, and this is a convenient place to do so. Is that the case?

Yes, I thought this would be one place we could detect an orphan.